src/CachePoolTest.php

```diff
@@ -520,7 +540,11 @@ public function testGetItemsInvalidKeys($key)
             return;
         }
 
-        $this->cache->getItems(['key1', $key, 'key2']);
+        $items = $this->cache->getItems(['key1', $key, 'key2']);
+
+        if ($items instanceof \Traversable) {
+            iterator_to_array($items);
```

Comment:
Using a generator is indeed not possible, as it would delay the validation of keys.
Or at least not a naive implementation of generators. It may be possible to write a generator-based API which would still validate keys upfront.